src/data/list/big_operators.lean

```diff
@@ -114,6 +114,17 @@ begin
   exact is_unit.mul (u h (mem_cons_self h t)) (prod_is_unit (λ m mt, u m (mem_cons_of_mem h mt)))
 end
 
+@[to_additive]
+lemma prod_is_unit_iff {α : Type*} [comm_monoid α] {L : list α} :
```

Comment:
Why does this need commutativity?

Reply:
Beacuse it's wrong otherwise: take the monoid of maps from the natural numbers to itself (under composition). Then if you first do x |-> x+1 and then x |-> x-1 (truncated sub), you get the identity, but neither map is invertible.